### ISSUE TITLE
Cannot read property 'connectTimeoutTimer' of undefined

### ISSUE THREAD

**Issue by 1ec5** [wikimedia/preq#18]
I get the following error when I attempt to run `gulp fetch-meta --portal` in the [wikimedia/portals](https://git.wikimedia.org/summary/wikimedia%2Fportals.git) project using node v5.8.0 and v6.2.2:

```
/path/to/portals/node_modules/preq/index.js:21
        if (this.options.connectTimeout && !s.connectTimeoutTimer) {
                                             ^

TypeError: Cannot read property 'connectTimeoutTimer' of undefined
    at ConnectTimeoutAgent.createSocket (/path/to/portals/node_modules/preq/index.js:21:46)
    at ConnectTimeoutAgent.Agent.addRequest (_http_agent.js:141:10)
    at new ClientRequest (_http_client.js:147:16)
    at Object.exports.request (http.js:31:10)
    at Object.exports.request (https.js:197:15)
    at Request.start (/path/to/portals/node_modules/request/request.js:746:30)
    at Request.end (/path/to/portals/node_modules/request/request.js:1357:10)
    at end (/path/to/portals/node_modules/request/request.js:574:14)
    at Immediate._onImmediate (/path/to/portals/node_modules/request/request.js:588:7)
    at processImmediate [as _immediateCallback] (timers.js:383:17)
```

[T128659](https://phabricator.wikimedia.org/T128659) may be relevant.

/cc @j4n-co

**Comment by lmessinger** [comment]
same here. seems that nodejs changed their http_agent implementation. preq code relies on it in the sense that it changes the default implementation of createSocket!

**Comment by Pchelolo** [comment]
#21 should fix this.

**Comment by jandre3000** [comment]
@1ec5 I've updated a related ticket with this error https://phabricator.wikimedia.org/T150190

**Comment by gwicke** [comment]
#21 is now merged & v0.5.0 is released. Closing.

**Comment by nyurik** [comment]
I keep seeing this error message in the code that should be totally unrelated -- the code that never uses preq, but calls request directly (3rd party lib). Is it possible that preq somehow hooks onto the request module, and gets called even when its not suppose to? Still investigating...
P.S. Partial confirmation:  the Kartotherian's "geoshapes" module resides on a very different express route, and is never used. But when it fails to initialize, it somehow affects a 3rd party tilelive-http module's requests for remote tiles (over http:) using request npm, which is used via a completely separate code path. I am not sure if this should be a separate bug - preq somehow affecting direct users of request npm.
Repo:
```bash
git clone https://github.com/kartotherian/kartotherian.git && cd kartotherian
git checkout af98c0a
sudo apt-get install libcairo2-dev libjpeg-dev libpango1.0-dev libgif-dev build-essential g++
npm install
node server.js -c config.external.yaml
```
* navigate to localhost:6533 - failure
* now, ctrl+c, and edit package.json - removing "@kartotherian/geoshapes" from the "requestHandlers"
* try again - this time it will work fine
